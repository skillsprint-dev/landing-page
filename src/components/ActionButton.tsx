"use client";
import { useUser } from "@clerk/nextjs";
import { useTranslation } from "react-i18next";
import "../../i18n.mjs";

export default function ActionButton() {
  const href = "https://calendly.com/skillsprint/free-assessment";
  const { t } = useTranslation();
  const { user } = useUser();
  const handleBookingClick = () => {
    if (!user) {
      window.location.href = '/sign-in';
    } else {
      window.location.href = href;
    }
  };

  return (
    <button
      onClick={handleBookingClick}
      className="inline-flex items-center justify-center w-full px-5 py-3 text-sm font-medium text-center text-gray-900 border border-gray-200 sm:w-auto hover:bg-gray-100 focus:ring-4 focus:ring-gray-100 dark:text-white dark:border-gray-700 dark:hover:bg-gray-700 dark:focus:ring-gray-800 hvr-sweep-to-right dark:bg-gray-700"
    >
      {t('actionButtonPage.bookAssessmentButton.text')}
    </button>
  )
}
