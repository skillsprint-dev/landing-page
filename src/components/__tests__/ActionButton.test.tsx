import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import ActionButton from '../ActionButton';

describe('ActionButton', () => {
    it('links to the Calendly page', () => {
        render(<ActionButton />);
        const actionButton = screen.getByRole('link', {
            name: 'Book a Free Assessment'
        });
        expect(actionButton).toHaveAttribute('href', 'https://calendly.com/skillsprint/free-assessment');
    });

    it('matches the snapshot', () => {
        const { asFragment } = render(<ActionButton />);
        expect(asFragment()).toMatchSnapshot();
    });
});