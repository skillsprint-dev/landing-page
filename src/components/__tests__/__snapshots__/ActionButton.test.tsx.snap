// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButton matches the snapshot 1`] = `
<DocumentFragment>
  <a
    class="inline-flex items-center justify-center w-full px-5 py-3 text-sm font-medium text-center text-gray-900 border border-gray-200 sm:w-auto hover:bg-gray-100 focus:ring-4 focus:ring-gray-100 dark:text-white dark:border-gray-700 dark:hover:bg-gray-700 dark:focus:ring-gray-800 hvr-sweep-to-right dark:bg-gray-700"
    href="https://calendly.com/skillsprint/free-assessment"
  >
    Book a Free Assessment
  </a>
</DocumentFragment>
`;
